resolver-dns/src/main/java/io/netty/resolver/dns/DnsQueryContext.java

```diff
@@ -60,9 +59,9 @@
         this.nameServerAddr = nameServerAddr;
         this.question = question;
         this.promise = promise;
-
-        id = allocateId();
         recursionDesired = parent.isRecursionDesired();
+        id = parent.queryContextManager.add(this);
```

Comment:
We should not do this as this allows this to escape from the constructor and so we may see not fully initialise instances .